File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    task = [i for i in range(len(priorities))]
    my_task = task[location]
    done = []
    
    while priorities:
        if priorities[0] != max(priorities):
            priorities.append(priorities.pop(0))
            task.append(task.pop(0))
        else:
            priorities.pop(0)
            done.append(task.pop(0))
            
    return done.index(my_task) + 1
            

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {

        // idx, 우선순위
        Queue<int[]> q = new ArrayDeque<>();

        int size = priorities.length;

        // 우선순위 리스트
        List<Integer> ps = new ArrayList<>();
        for (int n : priorities) ps.add(n);
        Collections.sort(ps, Collections.reverseOrder());

        // 큐에 넣기
        for (int i = 0; i < size; i++) {
            q.add(new int[] {i, priorities[i]});
        }

        int ans = 1;
        while (!q.isEmpty()) {
            int[] cur = q.poll();

            // 우선순위 검증
            // 만약 지금 우선순위가 더 뒤라면 다시 enqueue
            if (cur[1] < ps.get(0)) q.add(cur);
            else {
                if (cur[0] == location) break;
                ps.removeFirst();
                ans++;
            }
        }
        
        return ans;
    }
}